### PR TITLE
[INLONG-8586][Manager] Stop Stream Source which is still running after group is stopped

### LINKED ISSUE #8586: [Improve][Manager] Stop Stream Source which is still running after group is stopped
### Description

_No response_

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kipshi**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8586 

### Motivation

In extreme situation, some stream sources is still running after group is stopped or deleted.

### Modifications

Stop Stream Source which is still running after group is stopped with compromizer interface.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/group/compromize/Compromizer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package org.apache.inlong.manager.service.group.compromize;
+
+/**
+ * Operator which ensure the consistency of the state of all components within the group
+ */
+public interface Compromizer {
+
+    public abstract void compromize(String inlongGroupId);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/group/compromize/Compromizer.java:25]: 1. Non-default methods in interfaces do not require the `public abstract` modifier.
2. Please add javadoc for the method in interfaces, thanks.

image

Comment (kipshi) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/group/compromize/Compromizer.java]: done, removed

File: inlong-manager/manager-dao/src/main/resources/mappers/InlongGroupEntityMapper.xml [hunk 2]
```diff
@@ -390,20 +390,35 @@
...
+        <bind name="_isInlongService" value="LoginUser.InlongService"/>
+        select inlong_group_id
+        from inlong_group
+        <where>
+            modify_time >= #{timeAfter, jdbcType=TIMESTAMP}
+            <if test="_isInlongService == false">
+                and tenant = #{tenant,jdbcType=VARCHAR}
+            </if>
+        </where>
+        where
```
[the remark this answers is not in the document]
Comment (kipshi) [reply inlong-manager/manager-dao/src/main/resources/mappers/InlongGroupEntityMapper.xml]: done, removed

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/task/DelGroupCompromizerTask.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ * Stop all stream source which is running when group is deleted
+ */
+@Slf4j
+@Service
+public class DelGroupCompromizerTask extends TimerTask implements Compromizer, InitializingBean {
+
+    private static final int INITIAL_DELAY = 300;
+    private static final int INTERVAL = 1800;
+
+    @Value("${group.compromize.batchSize:100}")
```
[the remark this answers is not in the document]
Comment (kipshi) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/task/DelGroupCompromizerTask.java:55]: compromise,  spelling error

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-web/src/main/resources/application-prod.properties [hunk 4]
```diff
@@ -96,3 +96,6 @@ source.update.interval=60
 # If turned on, tasks in the incorrect state are periodically deleted
 source.cleansing.enabled=false
 source.cleansing.interval=600
+
+# Group batch size to compromise in one period
+group.compromise.batchSize=100
```
Comment (healchow) [inline inlong-manager/manager-web/src/main/resources/application-prod.properties:101]: This config is somewhat challenging to understand. Is it clearer to change it to this?
```
# Batch size of deleted inlong group ids
group.deleted.batchSize
```

**Review by healchow — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**